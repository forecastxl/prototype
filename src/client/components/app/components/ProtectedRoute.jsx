import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import React from 'react'
import { Route, Redirect } from 'react-router-dom'
import { selectors } from '../../../data/session'

const mapStateToProps = state => ({
  hasAuth: selectors.hasToken(state)
})

export function DumbProtectedRoute({ hasAuth, ...props }) {
  return hasAuth ? <Route {...props} /> : <Redirect to="/login" />
}

DumbProtectedRoute.propTypes = {
  hasAuth: PropTypes.bool.isRequired
}

export default connect(mapStateToProps)(DumbProtectedRoute)
