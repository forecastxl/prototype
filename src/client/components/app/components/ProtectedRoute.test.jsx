import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { DumbProtectedRoute } from './ProtectedRoute'

describe('<ProtectedRoute />', () => {
  it('renders a redirect when hasAuth is false', () => {
    const wrapper = shallow(
      <DumbProtectedRoute hasAuth={false} exact path="/" render={() => null} />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders the route when hasAuth is true', () => {
    const wrapper = shallow(<DumbProtectedRoute hasAuth exact path="/" render={() => null} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
