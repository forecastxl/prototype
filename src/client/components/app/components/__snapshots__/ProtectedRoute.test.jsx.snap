// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProtectedRoute /> renders a redirect when hasAuth is false 1`] = `
<Redirect
  push={false}
  to="/login"
/>
`;

exports[`<ProtectedRoute /> renders the route when hasAuth is true 1`] = `
<Route
  exact={true}
  path="/"
  render={[Function]}
/>
`;
